### PR TITLE
Various fixes

### PR DESCRIPTION
**Pull request by markdboyd**
## Changes proposed in this pull request:

- Add unit test case for empty response from UAA for user
- Refactoring
- Add script for running e2e test locally

## Security considerations

None

### CODE REVIEW

File: ci/init-config.sh [hunk 1]
```diff
@@ -1,7 +1,7 @@
 #!/usr/bin/env bash
 
 set -euo pipefail
-shopt -s inherit_errexit
```
Comment (bengerman13) [inline ci/init-config.sh:4]: is this to support older bash versions?

**Review by bengerman13 — APPROVED** [review]

**Review by bengerman13 — APPROVED** [review]
